docs: update spl token exchange integration (#19573)
solana-labs/solana: docs/src/integrations/exchange.md

method described above. Each new block should be scanned for successful transactions
issuing SPL Token [Transfer](https://github.com/solana-labs/solana-program-library/blob/fc0d6a2db79bd6499f04b9be7ead0c400283845e/token/program/src/instruction.rs#L105)
or [TransferChecked](https://github.com/solana-labs/solana-program-library/blob/fc0d6a2db79bd6499f04b9be7ead0c400283845e/token/program/src/instruction.rs#L268)
instructions referencing user accounts. It is possible that a transfer is initiated
by a smart contract via [Cross Program Invocation](/developing/programming-model/calling-between-programs#cross-program-invocations),
so [inner instructions](/terminology#inner-instruction) must be checked as well.
The `preTokenBalance` and `postTokenBalance` fields from the transaction's metadata
can then be used to determine the effective balance change.

### Withdrawing

The withdrawal address a user provides should be the that of their SOL wallet.

Before executing a withdrawal [transfer](#token-transfers),
the exchange should check the address as
[described above](#validating-user-supplied-account-addresses-for-withdrawals).

> Additionally it is recommended that the account at this address be owned by the
> System Program and that its SOL balance be non-zero.  Non-system owned withdrawal
> addresses should be rejected and zero-SOL-balance accounts should require user
> confirmation before proceeding.

Reviewer: ```suggestion
Additionally this address must be owned by the System Program and have no account data.  If the address has no SOL balance, user confirmation should be obtained before proceeding with the withdrawal.  All other withdrawal addresses must be rejected.
```
